packages/types/src/types.d.ts

```diff
@@ -11,4 +11,16 @@ export type AnyNumber = UInt | BN | Uint8Array | number | string;
 
 export type AnyU8a = U8a | Uint8Array | Array<number> | string;
 
+export interface Codec<T> {
+  byteLength(): number;
```

Comment:
We could, to get past the BN vs Array issue, call this `encodedLength`, which is actually exactly what it is. (As a property/getter)